src/classes/ToolboxClient.ts

```diff
@@ -140,9 +140,23 @@ export class ToolboxClient {
 		await this.writeUsernotes(subreddit, notes, reason);
 	}
 
-	/** */
+	/**
+	 * Retrieves toolbox configuration for a subreddit.
+	 * @param subreddit Name of the subreddit to retrieve config for
+	 */
 	async getConfig (subreddit: string) {
-		const page = await this.reddit.getWikiPage(subreddit, TB_CONFIG_PAGE);
-		return new SubredditConfig(page.content);
+		let page: WikiPage | undefined;
+		try {
+			page = await this.reddit.getWikiPage(subreddit, TB_CONFIG_PAGE);
+		} catch (error) {
+			// Devvit throws an error when page is not present.
+			// Check to see if page is actually absent, if it is present then
+			// rethrow error.
+			const allPages = await this.reddit.getWikiPages(subreddit);
+			if (allPages.includes(TB_CONFIG_PAGE)) {
+				throw error;
+			}
+		}
```

Comment:
The branches here are confusing to me - when is this expected to rethrow, and when is it expected to fall through with `page` undefined? I'm not familiar with when specifically `getWikiPage` throws, or what the actual criteria are for a page to be included in `getWikiPages`.

(I understand that leaving `page` undefined is fine since we get its content with `?.` below, I'm just not sure when exactly we expect each of these branches to be taken)

Reply:
The main situation I encounter where `getWikiPage` throws is where the page doesn't exist, but it's also theoretically possible for it to fail in other situations too. 

If the page doesn't exist, I figured it would probably be a good idea to check to see if the page _does_ exist. If it doesn't, then we can assume that the config hasn't been saved at all for a subreddit yet (and fall back on the default). But if it does exist, then there's another issue (permissions, or a Dev Plat issue) at play. I don't think it would be a good idea to assume the default config if a valid config page does exist because then invalid behaviour could result.

Reply:
Ah, thanks for explaining. Yeah, this seems like a good thing to check to me - I didn't consider that there could be other reasons for that call to throw. I'll suggest a clarification in the comment here:

```suggestion
		try {
			page = await this.reddit.getWikiPage(subreddit, TB_CONFIG_PAGE);
		} catch (error) {
			// Devvit throws an error when page is not present, but also
			// sometimes for other reasons. Check if the page actually
			// exists; if it doesn't we'll use the default config, but if it
			// does then something else is wrong and we'll rethrow the error.
			const allPages = await this.reddit.getWikiPages(subreddit);
			if (allPages.includes(TB_CONFIG_PAGE)) {
				throw error;
			}
		}
```